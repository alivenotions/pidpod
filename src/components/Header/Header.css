@import url('https://fonts.googleapis.com/css?family=Pacifico');

header {
  padding: 1em;
  margin: 0 1.5em;
  border-bottom: 1px solid #dde0dd;
}

header h1 {
  display: inline;
  font-family: 'Pacifico', sans-serif;
  padding-left: 15%;
  font-size: 3em;
}

header .add-podcast {
  margin: auto 0;
  position: absolute;
  right: 1em;
  top: 3em;
}

header button {
  font-size: 1em;
  padding: 0.4em;
  margin: 0 0.5em;
  font-weight: bold;
  height: 2.5em;
  width: 2.5em;
  border-radius: 50%;
  background: linear-gradient(45deg, #dde0dd 25%, #c4c4c4);
  box-shadow: 3px 1.5px 1px #cecfce;
  border: 1px solid #adafad;
  cursor: pointer;
}

header button:active {
  background: linear-gradient(45deg, #c4c4c4 25%, #dde0dd);
}

header button:disabled {
  background: #bebebe;
  opacity: 0.7;
  cursor: default;
  color: #807f7f;
  box-shadow: 0 0 0;
}

header input {
  font-size: 1em;
  /* width: 0; */
  padding: 0.4em;
  margin: 0 0.5em;
}
