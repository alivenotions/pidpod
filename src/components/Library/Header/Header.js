import React from 'react'

class Header extends React.Component {
  addPodcast = event => {
    event.preventDefault()
    // FIXME: Create a modal instead of an input type
    const rssUrl = document.getElementById('rssUrl').value
    console.log(rssUrl)
    this.props.addPodcast(rssUrl)
  }

  render() {
    return (
      <header>
        <h1>{this.props.heading}</h1>
        {/* TODO: Abstract this into a button component */}
        <input type="text" id="rssUrl" />
        <button onClick={this.addPodcast}>Add Podcast</button>
      </header>
    )
  }
}

export default Header
