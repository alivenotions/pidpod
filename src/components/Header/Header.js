import React, { Component } from 'react'

import './Header.css'

class Header extends Component {
  addPodcast = event => {
    event.preventDefault()
  }

  handleAddPodcastClick = event => {}

  render() {
    return (
      <header>
        <h1>{this.props.title}</h1>
        <div className="add-podcast">
          <input placeholder="Add RSS Feed" />
          <button>+</button>
        </div>
      </header>
    )
  }
}

export default Header
