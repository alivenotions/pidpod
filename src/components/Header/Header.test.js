import React from 'react'
import { shallow } from 'enzyme'

import Header from './Header'

describe('<Header />', () => {
  it('renders a header element', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper.find('header')).toHaveLength(1)
  })

  it('renders the title correctly', () => {
    const wrapper = shallow(<Header title="Tadow" />)
    expect(wrapper.find('header h1').text()).toEqual('Tadow')
  })

  it('should ensure that the button is disabled when input is empty', () => {
    const wrapper = shallow(<Header />)
    wrapper.setState({ rssUrlValue: '' })
    expect(wrapper.find('button').prop('disabled')).toEqual(true)
  })

  it('should call the prop function on button click', () => {
    const onButtonClick = jest.fn()
    const wrapper = shallow(<Header addPodcast={onButtonClick} />)
    wrapper.find('button').simulate('click', { preventDefault() {} })
    expect(onButtonClick).toHaveBeenCalled()
  })

  it('should call the prop function with input state on button click', () => {
    const onButtonClick = jest.fn()
    const wrapper = shallow(<Header addPodcast={onButtonClick} />)
    wrapper.setState({ rssUrlValue: 'mockRss.com/value/rss' })
    wrapper.find('button').simulate('click', { preventDefault() {} })
    expect(onButtonClick).toHaveBeenCalledWith('mockRss.com/value/rss')
  })
})
