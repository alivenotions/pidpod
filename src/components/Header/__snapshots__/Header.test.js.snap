// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders correctly 1`] = `
<header>
  <h1>
    PidPod
  </h1>
  <div
    className="add-podcast"
  >
    <input
      onChange={[Function]}
      placeholder="Add RSS Feed"
      type="text"
      value=""
    />
    <button
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
</header>
`;

exports[`<Header /> should call the prop function with input state on button click 1`] = `
<header>
  <h1>
    PidPod
  </h1>
  <div
    className="add-podcast"
  >
    <input
      onChange={[Function]}
      placeholder="Add RSS Feed"
      type="text"
      value="mockRss.com/value/rss"
    />
    <button
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
</header>
`;
